src/Roslyn.Diagnostics.Analyzers/CSharp/CSharpInvokeTheCorrectPropertyToEnsureCorrectUseSiteDiagnostics.cs

```diff
@@ -0,0 +1,89 @@
+                RoslynDiagnosticsAnalyzersResources.ResourceManager, typeof(RoslynDiagnosticsAnalyzersResources));
+
+        private static readonly LocalizableString s_localizableMessage =
+            new LocalizableResourceString(nameof(RoslynDiagnosticsAnalyzersResources.InvokeTheCorrectPropertyToEnsureCorrectUseSiteDiagnosticsMessage),
+                RoslynDiagnosticsAnalyzersResources.ResourceManager, typeof(RoslynDiagnosticsAnalyzersResources));
+
+        internal static readonly DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+            RoslynDiagnosticIds.UseSiteDiagnosticsCheckerRuleId,
+            s_localizableTitle,
+            s_localizableMessage,
+            "Usage",
+            DiagnosticSeverity.Error,
+            false,
+            null,
+            null,
+            WellKnownDiagnosticTags.Telemetry);
+
+        private static readonly ImmutableDictionary<string, string> s_propertiesToValidateMap =
+            new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
+            {
+                { s_baseTypeString, s_typeSymbolFullyQualifiedName},
+                { s_interfacesString, s_typeSymbolFullyQualifiedName},
+                { s_allInterfacesString, s_typeSymbolFullyQualifiedName},
+                { s_typeArgumentsString, s_namedTypeSymbolFullyQualifiedName},
+                { s_constraintTypesString, s_typeParameterSymbolFullyQualifiedName}
+            }.ToImmutableDictionary();
+
+        private const string s_baseTypeString = "BaseType";
+        private const string s_interfacesString = "Interfaces";
+        private const string s_allInterfacesString = "AllInterfaces";
+        private const string s_typeArgumentsString = "TypeArguments";
+        private const string s_constraintTypesString = "ConstraintTypes";
+
+        private const string s_typeSymbolFullyQualifiedName = "Microsoft.CodeAnalysis.CSharp.Symbols.TypeSymbol";
+        private const string s_namedTypeSymbolFullyQualifiedName = "Microsoft.CodeAnalysis.CSharp.Symbols.NamedTypeSymbol";
+        private const string s_typeParameterSymbolFullyQualifiedName = "Microsoft.CodeAnalysis.CSharp.Symbols.TypeParameterSymbol";
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterSyntaxNodeAction(AnalyzeNode, SyntaxKind.SimpleMemberAccessExpression);
+        }
+
+        private static void AnalyzeNode(SyntaxNodeAnalysisContext context)
+        {
+            var name = ((MemberAccessExpressionSyntax)context.Node).Name;
+
+            if (name.Kind() == SyntaxKind.IdentifierName)
+            {
+                var identifier = (IdentifierNameSyntax)name;
+                string containingTypeName = null;
```

Comment:
Out var syntax

Reply:
@333fred Hold off on review. Per Aleksey's feedback, I'm going to try and remove the dangerous properties, rather than use an analyzer to guard against them.
Thanks

Reply:
Alright, sounds good. I must have started looking before Aleksey's comment.